app/src/main/java/org/mozilla/fenix/experiments/NimbusSetup.kt

```diff
@@ -88,32 +95,26 @@ fun createNimbus(context: Context, url: String?): NimbusApi {
             // generated code.
             register(observer)
 
-            // This performs the minimal amount of work required to load branch and enrolment data
-            // into memory. If `getExperimentBranch` is called from another thread between here
-            // and the next nimbus disk write (setting `globalUserParticipation` or
-            // `applyPendingExperiments()`) then this has you covered.
-            // This call does its work on the db thread.
-            initialize()
+            val isFirstNimbusRun = if (context.settings().isFirstNimbusRun || url.isNullOrBlank()) {
+                setExperimentsLocally(R.raw.initial_experiments)
+                context.settings().isFirstNimbusRun = false
+                true
+            } else {
+                false
+            }
```

Comment:
I don't think we need this if-block. The `initialize` method below will set the experiments locally if it is the first nimbus run.

And `context.settings().isFirstNimbusRun = false` could be moved to after the `runBlocking` block below.

Reply:
but the problem will be that we will `context.settings().isFirstNimbusRun = false` every time, even if it was previously set

Reply:
We can still have the `isFirstNimbusRun` variable if needed for that

Reply:
Addressed :)